neuro_san/session/async_direct_agent_session.py

```diff
@@ -132,7 +132,7 @@ async def streaming_chat(self, request_dict: Dict[str, Any]) -> Generator[Dict[s
         user_input = extractor.get("user_message.text")
 
         # Create the gateway to the internals.
-        chat_session = DataDrivenChatSession(registry=self.tool_registry)
+        chat_session = DataDrivenChatSession(agent_network=self.agent_network)
```

Comment:
DataDrivenChatSession now takes an AgentNetwork as its arg.
The factory aspect of AgentToolRegistry is an implementation detail at this level now. (huzzah!).